crates/uv-auth/src/middleware.rs

```diff
@@ -40,102 +75,497 @@ impl Middleware for AuthMiddleware {
         extensions: &mut Extensions,
         next: Next<'_>,
     ) -> reqwest_middleware::Result<Response> {
-        let url = request.url().clone();
-
-        // If the request already has an authorization header, we don't need to do anything.
-        // This gives in-URL credentials precedence over the netrc file.
-        if request
-            .headers()
-            .contains_key(reqwest::header::AUTHORIZATION)
-        {
-            debug!("Credentials are already present for {url}");
-            return next.run(request, extensions).await;
-        }
+        // Check for credentials attached to (1) the request itself
+        let credentials = Credentials::from_request(&request);
+        // In the middleware, existing credentials are already moved from the URL
+        // to the headers so for display purposes we restore some information
```

Comment:
Not a huge fan of code that only runs in debug. Maybe we just do this every time (or never)?

Reply:
It's just for display, I wanted to avoid the extra clone on every request. It's really helpful for seeing what username we're processing and if there's a password. I think it's worth having for debugging and I'd want it on-always rather than removed.